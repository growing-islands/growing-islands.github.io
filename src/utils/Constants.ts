type BreakpointsType = {
  PC: number
  SP: number
}

export const BREAKPOINTS: BreakpointsType = {
  PC: 769,
  SP: 768,
}

export const PAGE_INFO = {
  URL: process.env.NEXT_PUBLIC_SITE_URL || 'https://',
  TWITTER: 'https://twitter.com/yumelab1214',
  FACEBOOK: 'https://www.facebook.com/yumelab.co.jp/',
}

export const INTERSECTION_OPTIONS = {
  root: null,
  rootMargin: '0px 0px -100%',
  threshold: 0,
}
