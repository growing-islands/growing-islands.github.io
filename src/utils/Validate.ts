/**
 * Form Validate関連の変数
 */

export const ERROR_MESSAGES = {
  required: 'この項目は入力必須です',
  maxLength: '入力できる文字数を超えています',
  mixLength: '文字数が足りません',
  max: '数値が大きすぎます',
  mix: '数値が小さすぎます',
  num: '半角数字で入力して下さい',
  hiragana: 'ひらがなで入力してください',
  fullKana: '全角カタカナで入力してください',
  tel: '電話番号が正しい形式ではありません',
  email: 'メールアドレスの形式が正しくありません',
  password:
    '英文字（大文字、小文字）、数字をそれぞれ最低一文字以上入れた8〜16文字で入力してください',
  dependent_char: '使用できない文字列が含まれています',
  url: '「http://」もしくは「https://」から始まるURLを入力してください',
}

// NOTE： 正規表現を記述するためにESLintを無効
/*eslint-disable no-useless-escape */
/*eslint-disable no-irregular-whitespace */
export const REGEX = {
  num: /^[0-9]+$/,
  alphanumeric: /^[0-9a-zA-Z]+$/,
  hiragana: /^[ぁ-んー　]*$/,
  fullKana: /^[ァ-ヶー　]*$/,
  halfKana: /^[ｦ-ﾟ]*$/,
  tel: /^(0{1}\d{9,10})$/,
  email:
    /^([\w!#$%&'*+\-\/=?^`{|}~]+(\.[\w!#$%&'*+\-\/=?^`{|}~]+)*|"([\w!#$%&'*+\-\/=?^`{|}~. ()<>\[\]:;@,]|\\[\\"])+")@(([a-zA-Z\d\-]+\.)+[a-zA-Z]+|\[(\d{1,3}(\.\d{1,3}){3}|IPv6:[\da-fA-F]{0,4}(:[\da-fA-F]{0,4}){1,5}(:\d{1,3}(\.\d{1,3}){3}|(:[\da-fA-F]{0,4}){0,2}))\])$/,
  password:
    /^(?=.*?[a-z])(?=.*?[A-Z])(?=.*?\d)[a-zA-Z0-9!"#$%&'()\*\+\-\.,\/:;<=>?@\[\\\]^_`{|}~]{8,16}$/,
  dependent_char:
    /^(?!.*(\u2460|\u2461|\u2462|\u2463|\u2464|\u2465|\u2466|\u2467|\u2468|\u2469|\u246a|\u246b|\u246c|\u246d|\u246f|\u2470|\u2471|\u2472|\u2473|\u2160|\u2161|\u2162|\u2163|\u2164|\u2165|\u2166|\u2167|\u2168|\u2169|\u3349|\u3314|\u3322|\u334d|\u3318|\u3327|\u3303|\u3336|\u3351|\u3357|\u330d|\u3326|\u3323|\u332b|\u334a|\u333b|\u339c|\u339d|\u339e|\u338e|\u338f|\u33c4|\u33a1|\u337b|\u301d|\u301f|\u2116|\u33cd|\u2121|\u32a4|\u32a5|\u32a6|\u32a7|\u32a8|\u3231|\u3232|\u3239|\u337e|\u337d|\u337c|\u222e|\u221f|\u22bf|\u2170|\u2171|\u2172|\u2173|\u2174|\u2175|\u2176|\u2177|\u2178|\u2179|\uffe2|\uffe4|[\uD800-\uDBFF][\uDC00-\uDFFF])).*$/,
  url: /^(https?)(:\/\/[-_.!~*\'()a-zA-Z0-9;\/?:\@&=+\$,%#]+)$/,
}
/*eslint-enable no-useless-escape */
/*eslint-enable no-irregular-whitespace */
