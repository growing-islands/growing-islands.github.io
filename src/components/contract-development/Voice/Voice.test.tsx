import { render } from '~tests/libs/utils'
import { Voice } from '.'

describe('components/contract-development/Voice', () => {
  it('Snapshot', () => {
    const { asFragment } = render(<Voice />)

    expect(asFragment()).toMatchSnapshot()
  })
})
