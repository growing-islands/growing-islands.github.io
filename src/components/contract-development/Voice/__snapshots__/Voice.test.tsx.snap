// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/contract-development/Voice Snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
    id="voice"
  >
    <div
      class="l-wrap"
    >
      <div
        class="l-inner"
      >
        <div
          class="hidden tablet:block"
          data-testid="component"
        >
          <div
            class=""
          >
            <div>
              <span
                translate="no"
              >
                Voice
              </span>
              <hr />
              <span>
                お客様の声
              </span>
            </div>
          </div>
        </div>
        <header
          data-aos="fade-up"
        >
          <h2>
            お客様の声
          </h2>
        </header>
        <ul>
          <li
            data-aos="fade-up"
          >
            <div
              class=""
            >
              <div>
                <div>
                  <img
                    layout="responsive"
                    placeholder="blur"
                    src="test-file-stub"
                  />
                </div>
              </div>
              <div>
                <h3
                  class=""
                  data-testid="component"
                >
                  自社サービスを無駄なコストをかける事なく開発
                </h3>
                <p
                  class=""
                  data-testid="component"
                >
                  有限会社スズキプリンティングサービス様
自社サービス開発
                </p>
                <p
                  class=""
                  data-testid="component"
                >
                  自社サービスの開発を依頼し、開発期間が1年と長期に渡る中、やはり要件の変更なども柔軟に対応できるユメラボさんの「共創する受託開発」で良かったと思える分岐がたくさんありました。開発開始までの期間も短く、月額の初期投資も抑えながら、必要な機能作成と検証を繰り返す事で、結果的に無駄なコストをかける事なく開発できたと感じています。
                </p>
              </div>
            </div>
          </li>
          <li
            data-aos="fade-up"
          >
            <div
              class=""
            >
              <div>
                <div>
                  <img
                    layout="responsive"
                    placeholder="blur"
                    src="test-file-stub"
                  />
                </div>
              </div>
              <div>
                <h3
                  class=""
                  data-testid="component"
                >
                  プロジェクトの途中からでも柔軟に対応
                </h3>
                <p
                  class=""
                  data-testid="component"
                >
                  人材サービス系企業様
自社サービス開発
                </p>
                <p
                  class=""
                  data-testid="component"
                >
                  自社のSaaSに関して開発途中から参画頂きました。
市場を取り巻く環境が変わる中、月額制の顧問契約で開発の要件変更にも対応いただき、インフラからフロントまで幅広く対応いただきました。
週に1回の定例ミーティングで進捗のすり合わせを行い、ディレクターの方に柔軟な対応を頂いたのが特に印象的です。
公開後もPDCAを回しながら、マーケティングの観点からも様々な相談ができるので、とても心強いと感じています。
                </p>
              </div>
            </div>
          </li>
          <li
            data-aos="fade-up"
          >
            <div
              class=""
            >
              <div>
                <div>
                  <img
                    layout="responsive"
                    placeholder="blur"
                    src="test-file-stub"
                  />
                </div>
              </div>
              <div>
                <h3
                  class=""
                  data-testid="component"
                >
                  定額制でDXの第一歩からサポート
                </h3>
                <p
                  class=""
                  data-testid="component"
                >
                  食品系メーカー様
社内システム開発
                </p>
                <p
                  class=""
                  data-testid="component"
                >
                  社内で使うシステムの構築を依頼しました。
別の会社に見積もりを取った際、最終的に一括でまとまった金額が必要との話になり難しい判断を迫られていた所、月額制で対応していただけるユメラボさんにお話を聞ける機会がありました。
毎月定額で契約できる所が予算を取りやすく、また、オフショア開発との事で満足できる価格感で進んでおります。
内容的も定例の際に丁寧な対応を頂き、スケジュール通り機能ができて行くので安心しています。
                </p>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
